comp/core/secrets/secrets.go

```diff
@@ -53,14 +49,53 @@ type secretContext struct {
 	yamlPath string
 }
 
-func init() {
-	secretCache = make(map[string]string)
-	secretOrigin = make(handleToContext)
+func newSecretResolverProvider(deps dependencies) provides {
+	resolver := &secretResolver{
+		cache:           make(map[string]string),
+		origin:          make(handleToContext),
+		responseMaxSize: 1024 * 1024,
+	}
+
+	// TODO: hack to save a reference to the singleton instance
+	if instance == nil {
+		instance = resolver
+	}
+
+	return provides{
+		Comp:          resolver,
+		FlareProvider: resolver.FlareProvider(),
+	}
 }
 
-func registerSecretOrigin(handle string, origin string, yamlPath []string) {
+var instance *secretResolver
+
+func GetInstance() Component {
+	// TODO: temporary hack to allow non-componentized code to use the secrets resolver
+	if instance == nil {
+		p := newSecretResolverProvider(dependencies{})
+		instance = p.Comp.(*secretResolver)
+	}
+	return instance
+}
```

Comment:
This seems to be used only in `cmd/agent/subcommands/` and `pkg/diagnose/` could we pass secret from the main to those part and remove the singleton ?

Reply:
I was able to remove some of the uses of instance, but I think it may be necessary for now. The difficult one is here: https://github.com/DataDog/datadog-agent/pull/20883/files#diff-c792372d6a3476ee5cf3099091bfd5ce07d3f7223bd4d4627a75f88444a380b8R101

In order to add the secrets component to the caller, we would need to change the signature for the `diagnose` function, which would be a large change that would be difficult to undo later, and nearly every other usage does not care about the secrets component.

The thing that's unclear to me is how should the diagnose command be getting components that it needs? Will it be turned into a component itself?